crates/xilem_masonry/src/view/button.rs

```diff
@@ -53,4 +44,22 @@ where
             ChangeFlags::UNCHANGED
         }
     }
+
+    fn message(
+        &self,
+        _view_state: &mut Self::ViewState,
+        id_path: &[ViewId],
+        message: Box<dyn std::any::Any>,
+        app_state: &mut State,
+    ) -> crate::MessageResult<Action> {
+        debug_assert!(
+            id_path.is_empty(),
+            "id path should be empty in Button::message"
+        );
+        if let Some(masonry::Action::ButtonPressed) = message.downcast_ref() {
```

Comment:
To be consistent, I'd probably use `downcast` here; other views likely will want to take ownership of the action, and this box will be immediately dropped in the case where this suceeds anyway

Reply:
The reason I'm using `downcast_ref` instead of `downcast`, is that then `MessageResult::Stale(message)` is possible. Otherwise I need to do something like `MessageResult::Stale(Box::new(()))` if that's better?

Reply:
Oh, wait, `downcast` returns `Self` in `Err`, ok, then I'm changing it.